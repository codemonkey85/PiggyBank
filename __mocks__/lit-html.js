"use strict"

const jestConfig = require("../jest.config")

const litHtml = {
  render: jest.fn(),
  html: jest.fn()
}

module.exports = litHtml