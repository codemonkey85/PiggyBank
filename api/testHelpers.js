const testHelpers = {
  normalize: (text) => {
    return text.replace(/\s/g, "").toLowerCase();
  },
  mockRouter: () => {
    return {
      get: jest.fn(),
      post: jest.fn(),
      put: jest.fn(),
      delete: jest.fn()
    };
  },
  mockRepo: () => {
    return {
      account: {
        selectAll: jest.fn(),
        insert: jest.fn(),
        update: jest.fn(),
        delete: jest.fn()
      }
    };
  },
  mockRequest: () => {

  },
  mockResult: () => {
    return {
      json: jest.fn()
    };
  }
};

module.exports = testHelpers;