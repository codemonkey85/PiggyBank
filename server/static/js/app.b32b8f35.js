(function(e){function t(t){for(var n,u,l=t[0],s=t[1],i=t[2],c=0,v=[];c<l.length;c++)u=l[c],Object.prototype.hasOwnProperty.call(o,u)&&o[u]&&v.push(o[u][0]),o[u]=0;for(n in s)Object.prototype.hasOwnProperty.call(s,n)&&(e[n]=s[n]);p&&p(t);while(v.length)v.shift()();return a.push.apply(a,i||[]),r()}function r(){for(var e,t=0;t<a.length;t++){for(var r=a[t],n=!0,u=1;u<r.length;u++){var s=r[u];0!==o[s]&&(n=!1)}n&&(a.splice(t--,1),e=l(l.s=r[0]))}return e}var n={},o={app:0},a=[];function u(e){return l.p+"../static/js/"+({about:"about"}[e]||e)+"."+{about:"7f2f46da"}[e]+".js"}function l(t){if(n[t])return n[t].exports;var r=n[t]={i:t,l:!1,exports:{}};return e[t].call(r.exports,r,r.exports,l),r.l=!0,r.exports}l.e=function(e){var t=[],r=o[e];if(0!==r)if(r)t.push(r[2]);else{var n=new Promise((function(t,n){r=o[e]=[t,n]}));t.push(r[2]=n);var a,s=document.createElement("script");s.charset="utf-8",s.timeout=120,l.nc&&s.setAttribute("nonce",l.nc),s.src=u(e);var i=new Error;a=function(t){s.onerror=s.onload=null,clearTimeout(c);var r=o[e];if(0!==r){if(r){var n=t&&("load"===t.type?"missing":t.type),a=t&&t.target&&t.target.src;i.message="Loading chunk "+e+" failed.\n("+n+": "+a+")",i.name="ChunkLoadError",i.type=n,i.request=a,r[1](i)}o[e]=void 0}};var c=setTimeout((function(){a({type:"timeout",target:s})}),12e4);s.onerror=s.onload=a,document.head.appendChild(s)}return Promise.all(t)},l.m=e,l.c=n,l.d=function(e,t,r){l.o(e,t)||Object.defineProperty(e,t,{enumerable:!0,get:r})},l.r=function(e){"undefined"!==typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(e,"__esModule",{value:!0})},l.t=function(e,t){if(1&t&&(e=l(e)),8&t)return e;if(4&t&&"object"===typeof e&&e&&e.__esModule)return e;var r=Object.create(null);if(l.r(r),Object.defineProperty(r,"default",{enumerable:!0,value:e}),2&t&&"string"!=typeof e)for(var n in e)l.d(r,n,function(t){return e[t]}.bind(null,n));return r},l.n=function(e){var t=e&&e.__esModule?function(){return e["default"]}:function(){return e};return l.d(t,"a",t),t},l.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},l.p="",l.oe=function(e){throw console.error(e),e};var s=window["webpackJsonp"]=window["webpackJsonp"]||[],i=s.push.bind(s);s.push=t,s=s.slice();for(var c=0;c<s.length;c++)t(s[c]);var p=i;a.push([0,"chunk-vendors"]),r()})({0:function(e,t,r){e.exports=r("56d7")},"034f":function(e,t,r){"use strict";r("85ec")},1182:function(e,t,r){},"34ba":function(e,t,r){"use strict";r("1182")},"35e6":function(e,t,r){"use strict";r("94e5")},"56d7":function(e,t,r){"use strict";r.r(t);r("e260"),r("e6cf"),r("cca6"),r("a79d");var n=r("2b0e"),o=function(){var e=this,t=e.$createElement,r=e._self._c||t;return r("div",{attrs:{id:"app"}},[r("nav-bar"),r("router-view")],1)},a=[],u=function(){var e=this,t=e.$createElement,r=e._self._c||t;return r("nav",{attrs:{role:"navigation"}},[r("router-link",{attrs:{to:"/"}},[e._v("Home")]),e._v(" | "),r("router-link",{attrs:{to:"/about"}},[e._v("About")])],1)},l=[],s={name:"NavBar"},i=s,c=(r("34ba"),r("2877")),p=Object(c["a"])(i,u,l,!1,null,null,null),v=(p.exports,{name:"App",components:{NavBar:NavBar}}),f=v,h=(r("034f"),Object(c["a"])(f,o,a,!1,null,null,null)),g=h.exports,m=(r("d3b7"),r("3ca3"),r("ddb0"),r("8c4f")),_=function(){var e=this,t=e.$createElement,n=e._self._c||t;return n("div",{staticClass:"home"},[n("img",{attrs:{alt:"Vue logo",src:r("cf05")}}),n("HelloWorld",{attrs:{msg:"Welcome to Your Vue.js App"}})],1)},b=[],d=function(){var e=this,t=e.$createElement,r=e._self._c||t;return r("div",{staticClass:"hello"},[r("h1",[e._v(e._s(e.msg))]),e._m(0),r("h3",[e._v("Installed CLI Plugins")]),e._m(1),r("h3",[e._v("Essential Links")]),e._m(2),r("h3",[e._v("Ecosystem")]),e._m(3)])},j=[function(){var e=this,t=e.$createElement,r=e._self._c||t;return r("p",[e._v(" For a guide and recipes on how to configure / customize this project,"),r("br"),e._v(" check out the "),r("a",{attrs:{href:"https://cli.vuejs.org",target:"_blank",rel:"noopener"}},[e._v("vue-cli documentation")]),e._v(". ")])},function(){var e=this,t=e.$createElement,r=e._self._c||t;return r("ul",[r("li",[r("a",{attrs:{href:"https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-babel",target:"_blank",rel:"noopener"}},[e._v("babel")])]),r("li",[r("a",{attrs:{href:"https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-router",target:"_blank",rel:"noopener"}},[e._v("router")])]),r("li",[r("a",{attrs:{href:"https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-eslint",target:"_blank",rel:"noopener"}},[e._v("eslint")])]),r("li",[r("a",{attrs:{href:"https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-unit-jest",target:"_blank",rel:"noopener"}},[e._v("unit-jest")])])])},function(){var e=this,t=e.$createElement,r=e._self._c||t;return r("ul",[r("li",[r("a",{attrs:{href:"https://vuejs.org",target:"_blank",rel:"noopener"}},[e._v("Core Docs")])]),r("li",[r("a",{attrs:{href:"https://forum.vuejs.org",target:"_blank",rel:"noopener"}},[e._v("Forum")])]),r("li",[r("a",{attrs:{href:"https://chat.vuejs.org",target:"_blank",rel:"noopener"}},[e._v("Community Chat")])]),r("li",[r("a",{attrs:{href:"https://twitter.com/vuejs",target:"_blank",rel:"noopener"}},[e._v("Twitter")])]),r("li",[r("a",{attrs:{href:"https://news.vuejs.org",target:"_blank",rel:"noopener"}},[e._v("News")])])])},function(){var e=this,t=e.$createElement,r=e._self._c||t;return r("ul",[r("li",[r("a",{attrs:{href:"https://router.vuejs.org",target:"_blank",rel:"noopener"}},[e._v("vue-router")])]),r("li",[r("a",{attrs:{href:"https://vuex.vuejs.org",target:"_blank",rel:"noopener"}},[e._v("vuex")])]),r("li",[r("a",{attrs:{href:"https://github.com/vuejs/vue-devtools#vue-devtools",target:"_blank",rel:"noopener"}},[e._v("vue-devtools")])]),r("li",[r("a",{attrs:{href:"https://vue-loader.vuejs.org",target:"_blank",rel:"noopener"}},[e._v("vue-loader")])]),r("li",[r("a",{attrs:{href:"https://github.com/vuejs/awesome-vue",target:"_blank",rel:"noopener"}},[e._v("awesome-vue")])])])}],k={name:"HelloWorld",props:{msg:String}},y=k,w=(r("35e6"),Object(c["a"])(y,d,j,!1,null,"27d51c9c",null)),O=w.exports,x={name:"Home",components:{HelloWorld:O}},E=x,P=Object(c["a"])(E,_,b,!1,null,null,null),$=P.exports;n["a"].use(m["a"]);var C=[{path:"/",name:"Home",component:$},{path:"/about",name:"About",component:function(){return r.e("about").then(r.bind(null,"f820"))}}],H=new m["a"]({mode:"history",base:"",routes:C}),S=H;n["a"].config.productionTip=!1,new n["a"]({router:S,render:function(e){return e(g)}}).$mount("#app")},"85ec":function(e,t,r){},"94e5":function(e,t,r){},cf05:function(e,t,r){e.exports=r.p+"../static/img/logo.82b9c7a5.png"}});
//# sourceMappingURL=app.b32b8f35.js.map