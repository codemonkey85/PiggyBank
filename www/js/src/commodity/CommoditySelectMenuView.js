"use strict"

import { html, render } from "../../lib/lit-html/lit-html.js"
import { getUuid } from "../PiggyBankUtil.js"

const template = (d) => {
  d.uuid = getUuid()
  return html`
    <label for='selectCommodityId-${d.uuid}'>${d.label}</label>
    <select id='selectCommodityId-${d.uuid}' class='select-commodity-id form-control'>
      <option value=''>[Choose Commodity]</option>
    </select>
`}

export class CommoditySelectMenuView extends Backbone.View {
  render(commodityId, label) {
    label = label || "Commodity"
    render(template({ label }), this.el)
    this.addCommodityOptions(window.piggybank.commodities.models, commodityId)
    return this
  }

  addCommodityOptions(collection, commodityId) {
    const commoditySelect = this.$(".select-commodity-id")
    for (let model of collection) {
      const option = `<option value='${model.get("id")}'>${model.get("name")}</option>`
      commoditySelect.append(option)
    }

    commoditySelect.val(commodityId)
    commoditySelect.on('change', (e) => this.trigger('change', e))
  }
}
