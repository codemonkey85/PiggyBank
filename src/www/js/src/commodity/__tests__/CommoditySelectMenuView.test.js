import "../../__tests__/testHelpers"
import { html, render } from "../../../lib/lit-html/lit-html.js"
import { CommoditySelectMenuView } from "../CommoditySelectMenuView"

test("CommoditySelectMenuView has render method", () => {
  const model = {
    get: jest.fn()
  }
  const collection = {
    models: [model]
  }
  const view = new CommoditySelectMenuView(collection)
  const renderedView = view.render()
  expect(view).toBe(renderedView)
  expect(html).toMatchSnapshot()
})