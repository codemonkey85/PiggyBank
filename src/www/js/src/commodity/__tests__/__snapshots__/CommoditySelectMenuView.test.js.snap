// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommoditySelectMenuView has render method 1`] = `[Function]`;
